src/web_app/web/reminder_view.py

```diff
@@ -0,0 +1,136 @@
+
+app_reminder = Blueprint('app_reminder', __name__, template_folder='templates')
+logger = Logger(log_path)
+reminder_manager = ReminderManager(client, DataBase(), logger)  # used for all interactions with database
+
+@app_reminder.route('/ui/reminder', methods=['GET'])
+def get_reminders():
+	"""
+	default mapping with all reminders
+	"""
+	reminders = reminder_manager.get_reminders()
+	courses, authors = reminder_manager.get_filter_options()
+	return render_template('reminder.html', courses=courses, authors=authors, reminders=reminders), 200
+
+
+@app_reminder.route('/ui/reminder/filter', methods=['POST'])
+def filter_reminders():
+	"""
+	mapping for returning only filtered reminders
+	params: name : name of the course
+			author : first and last name of the author
+			from : all dates after it
+			to : all dates after it
+			posted : true or false, reminder posted in workspace or not
+	returns: dict that has form for loading bootstrap table
+			result : number of reminders
+			rows : all data for reminders, must be JSON serializable
+	"""
+	filters = dict()
+	if len(request.args) != 0:
+		for elem in ['name', 'author', 'from', 'to', 'posted']:
+			filters[elem] = request.args.get(elem)
+
+		# getting rid of not selected filters
+		filters.pop('name') if filters['name'] == "Nothing selected" else None
+		filters.pop('author') if filters['author'] == "Nothing selected" else None
+		filters.pop('from') if not filters['from'] else None    # default is empty string
+		filters.pop('to') if not filters['to'] else None
+
+		reminders = reminder_manager.get_reminders(**filters)
+
+	# canceling filters and returning all reminders
+	else:
+		reminders = reminder_manager.get_reminders()
+
+	reminders_json = list()
+	# turn a list of reminders into a list of reminder attributes in dict form
+	# end_date and timer are parsed to string because timestamp and interval are not json serializable
+	for reminder in reminders:
+		reminders_json.append({'id': reminder.id,
+		                       'end_date': str(reminder.end_date),
+		                       'timer': str(reminder.timer),
+		                       'text': reminder.text,
+		                       'posted': reminder.posted})
+	return {'result': len(reminders_json), 'rows': reminders_json}
+
+
+@app_reminder.route('/ui/reminder/save', methods=['POST'])
+def save_reminder():
+	"""
+	mapping for saving edited reminder data to database
+	params: id : id of reminder
+			end_date : that of the event
+			timer : time left until end_date
+			text : description of the reminder
+			posted : reminder posted in workspace or not
+	returns: 400 if new data doesn't pass validation
+			 500 in case of database error
+			 200 otherwise
+	"""
+	reminder_data = dict()
+	for elem in ['id', 'end_date', 'timer', 'text', 'posted']:
+		if elem == 'timer':
+			# default format includes string "days", remove it and split data by :
+			reminder_data[elem] = request.args.get(elem).replace(' days, ', ':')
```

Comment:
ovo se ne može očistit prije samog slanja? tj da se u js malo urede podaci pa da na back dođu kolko tolko čisti bez potrebe za brisanjem stvari iz njih

Reply:
Vjerojatno može, budem pogledao to